OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -51,7 +52,7 @@ public void Killed(Actor self, AttackInfo e)
 				return;
 
 			var warhead = e.Warhead as DamageWarhead;
-			if (info.DeathType != null && warhead != null && !warhead.DamageTypes.Overlaps(info.DeathType))
+			if (warhead != null && !warhead.DamageTypes.Overlaps(info.DeathType))
```

Comment:
The problem is that the whole check - including the return - is supposed to be skipped if no DeathType is specified (see how in the old code, DeathType == null would skip the rest of the check including the return). 
The way the check is now, it will instead `return` if DeathType is empty (which is the default), because there's nothing to overlap.

Basically, this changes the default behavior from "ignore this check and continue with the following lines" to "return".